module/validation/receipt_validator.go

```diff
@@ -215,6 +215,11 @@ func (v *receiptValidator) validate(receipt *flow.ExecutionReceipt, getPreviousR
 		return fmt.Errorf("staked node invalid: %w", err)
 	}
 
+	prevResult, err := getPreviousResult(&receipt.ExecutionResult)
+	if err != nil {
+		return err
+	}
+
 	err = v.verifySignature(receipt, identity)
```

Comment:
moved up from below to skip validating the signature if the receipts is pending, in which case, we will postpone the verification until it becomes non-pending receipt.